subiquity/server/ad_joiner.py

```diff
@@ -28,19 +28,37 @@
 log = logging.getLogger('subiquity.server.ad_joiner')
 
 
+def debconf_disconnect():
+    """Disconnect from debconf. This is only to be used as a subprocess
+    preexec_fn helper."""
+    os.environ['DEBIAN_FRONTEND'] = 'noninteractive'
```

Comment:
Thanks for the early feedback. I'm looking at that now.

Reply:
af6f4177c62ee17148d36fc2a3cfcaf5818f2234 is working consistently. But not being happy with the implementation changes, I've being iterating over variations of this (with chroot and with curtin) since last night and couldn't get a stable way to reproduce the `pam-auth-update` issue I've seen.

Reply:
We point to `curtin in-target` as the solution for chrooting for [late-commands](https://ubuntu.com/server/docs/install/autoinstall-reference#late-commands), if something about that isn't working then we want a fix for that since someone doing a similar operation in late-commands would have the same problem.

Reply:
I'll recreate my virtual lab and explore more on this later. For now I'm convinced the only lasting issue is captured and fixed in https://github.com/canonical/subiquity/pull/1627 .

If I figure something out later, it will be a new issue/bug.